Prepare 0.22.1 release

mtreinish opened the pull request:
<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add a release note file using reno if the change needs to be
  documented in the release notes.
⚠️ If your pull request fixes an open issue, please link to the issue.

- [ ] I have added the tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
-->

### Summary

This commit prepares the 0.22.1 release, this consists of bumping the version number to 0.22.1. It also adds a prelude release note to provide a brief introduction to the release. When this commit merges it should be tagged as the 0.22.1 release.

### Details and comments

TODO:

- [x] Review and update release notes
- [x] Wait for last bugfix PRs to merge and be backported

jakelishman commented on releasenotes/notes/fix-pauli-basis-dep-27c0a4506ad38d2c.yaml:
```diff
@@ -2,10 +2,16 @@
     The ``pauli_list`` kwarg of :func:`.pauli_basis` has been deprecated as
-    :func:`.pauli_basis` now always returns a :class:`.PauliList`.
+    :func:`.pauli_basis` now always returns a :class:`.PauliList`. This argument
+    was removed pre-maturely from Qiskit Terra 0.22.0 which broke compatibility
```
```suggestion
    was removed prematurely from Qiskit Terra 0.22.0 which broke compatibility
```

jakelishman commented on releasenotes/notes/fix-pauli-basis-dep-27c0a4506ad38d2c.yaml:
```diff
@@ -2,10 +2,16 @@
-    broke existing code using the ``pauli_list=True`` future compatibility on
-    upgrade to Qiskit Terra 0.22. This kwarg has been added back to the the
-    function with a deprecation warning instead.
+    Fixed the pre-mature removal of the ``pauli_list`` keyword argument of
```
```suggestion
    Fixed the premature removal of the ``pauli_list`` keyword argument of
```

jakelishman commented on releasenotes/notes/fix-vqd-kgt2-1ed95de3e32102c1.yaml:
```diff
@@ -1,5 +1,7 @@
-    Fixed the :class:`~.eigensolvers.VQD` if more than ``k=2`` eigenvalues are computed.
-    Previously the code failed due to an internal type mismatch, but now runs as expected.
+    Fixed an issue with the :class:`~.eigensolvers.VQD` class if more than
+    ``k=2`` eigenvalues are computed. Previously this would fail due to an
```
```suggestion
    ``k=2`` eigenvalues were computed. Previously this would fail due to an
```

jakelishman reviewed (commented):
Strictly we're missing a full stop at the end of every `Fixed ...` link, but I don't know how much we care about fix that.

jakelishman reviewed (approved):
I'm happy whether or not we add the full stops.

mtreinish commented:
All the open PRs targeting 0.22.1 have merged, so I've removed the `on hold` label and added `automerge`. Once this merges we can tag the 0.22.1 release

mtreinish commented:
There was a CI issue last night that prevented the jobs used to build and publish the release artifacts from running. I manually ran them this morning so everything should be fine now if you re-run CI or update your branch.
